### DISCUSSION TITLE
cms-strapi example with graphql "mutation"

### DISCUSSION THREAD

**Discussion by onsombal** [vercel/next.js#15273] [Help] [answered]
Can anyone provide example Next.js code for the graphql mutation shown below? This mutation works perfectly in graphql playground.

**MUTATION — ADD AUTHOR WITH QUERY VARIABLES**
`mutation createAuthor($input: createAuthorInput!) {      
               createAuthor(input: $input) {  
                    author {  
                          name  
                    }  
               }  
}  `

**QUERY VARIABLES**
`{  
     "input": {  
         "data": {  
            "name": "Sam"        
         }  
     }   
}  `

**RESULT**
`{  
  "data": {  
    "createAuthor": {  
      "author": {  
        "name": "Sam"  
      }  
    }  
  }  
}  
`

**Comment by leerob** [comment]
Hey! This doesn't seem to be Next.js specific. Any GraphQL client that supports mutations should allow you to run this code. Maybe you could elaborate more on how Next specifically plays into this? Are you trying to use Apollo, graphql-request, etc?

**Comment by onsombal** [comment, marked as answer]
It's "Next.js specific" in the sense that the "cms-strapi" example employs Next.js. [https://github.com/vercel/next.js/tree/canary/examples/cms-strapi]

This sample app creates a blog, but only offers graphql "queries" as examples for querying authors and posts by authors. Strapi offers its own graphql server and resolvers that are based off of Apollo, but Apollo is not required.

I've been stumbling around trying to code a graphql "mutation" to add an author and posts. And, while my graphql code works well in the graphql playground, I'm struggling to convert it to code in the Next.js cms-strapi example.

Thoughts?

**Reply by leerob** [reply]
I'd recommend reading up on GraphQL mutations, and possibly using a client like [graphql-request](https://github.com/prisma-labs/graphql-request) to simplify this for you. Here's an (untested) attempt.

```js
export async function createAuthor(name) {
    const data = await fetchAPI(`
      mutation createAuthor($input: createAuthorInput!) { 
        createAuthor(input: $input) {
          author {
              name
          } 
        }
      }
  `, {
        variables: {
            input: {
            	data: {
                  name
            },
        }
    })
    return data?.createAuthor
}
```

**Reply by onsombal** [reply]
Thank you, Lee. I'll check it out and test!!!
